sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -47,6 +47,13 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
       plan.find(PlanHelper.specialExpressionsInUnsupportedOperator(_).nonEmpty).isEmpty)
   }
 
+  override protected val blacklistedOnceBatches: Set[String] =
+    Set("Pullup Correlated Expressions",
+      "Join Reorder",
+      "Subquery",
+      "Extract Python UDFs"
```

Comment:
Please create an umbrella JIRA and create a separate JIRA for each rule?

Reply:
Sure.